odl/discr/grid.py

```diff
@@ -638,33 +640,54 @@ def insert(self, index, other):
             [-6.0, 15.0],
             [-1.0, 0.0, 2.0]
         )
+        >>> g1.insert(1, g2, g2)
+        RectGrid(
+            [0.0, 1.0],
+            [1.0],
+            [-6.0, 15.0],
+            [1.0],
+            [-6.0, 15.0],
+            [-1.0, 0.0, 2.0]
+        )
 
         See Also
         --------
         append
         """
-        if not isinstance(other, RectGrid):
-            raise TypeError('`other` must be a `RectGrid` instance, got {}'
-                            ''.format(other))
-        idx = safe_int_conv(index)
-        # Support backward indexing
-        if idx < 0:
-            idx = self.ndim + idx
-        if not 0 <= idx <= self.ndim:
-            raise IndexError('index {} outside the valid range 0 ... {}'
-                             ''.format(idx, self.ndim))
-
-        new_vecs = (self.coord_vectors[:idx] + other.coord_vectors +
-                    self.coord_vectors[idx:])
-        return RectGrid(*new_vecs)
-
-    def append(self, other):
-        """Insert grid ``other`` at the end.
+        for i, grid in enumerate(grids):
+            if not isinstance(grid, RectGrid):
+                raise TypeError('all `grids` must be `RectGrid` instances, '
+                                'got {!r} as argument {}'
+                                ''.format(grid, i + 1))
+
+        index, index_in = safe_int_conv(index), index
+        if not -self.ndim <= index <= self.ndim:
+            raise IndexError('index {0} outside the valid range -{1} ... {1}'
+                             ''.format(index_in, self.ndim))
+        if index < 0:
+            index += self.ndim
+
+        if len(grids) > 1:
+            return self.insert(index, grids[0]).insert(
```

Comment:
Yes I realized that but put it off, thinking "nah, n is small". But we never know what people may come up with, so I'll fix it anyway.